File: apps/mybookkeeper/frontend/src/shared/types/applicant/applicant-detail-response.ts
import type { ApplicantEvent } from "./applicant-event";
import type { ApplicantReference } from "./applicant-reference";
import type { ApplicantStage } from "./applicant-stage";
import type { ScreeningResult } from "./screening-result";
import type { VideoCallNote } from "./video-call-note";

/**
 * Mirrors backend ``ApplicantDetailResponse`` — the full applicant detail
 * shape with all 1:N children nested.
 *
 * PII fields (``legal_name``, ``dob``, ``employer_or_hospital``,
 * ``vehicle_make_model``) come over the wire as plaintext — the backend's
 * ``EncryptedString`` TypeDecorator decrypts on read. The frontend renders
 * them behind a sensitive-unlock toggle so they're hidden by default per
 * RENTALS_PLAN.md §9.1.
 */
export interface ApplicantDetailResponse {
  id: string;
  organization_id: string;
  user_id: string;
  inquiry_id: string | null;

  legal_name: string | null;
  dob: string | null;
  employer_or_hospital: string | null;
  vehicle_make_model: string | null;
  id_document_storage_key: string | null;

  contract_start: string | null;
  contract_end: string | null;
  smoker: boolean | null;
  pets: string | null;
  referred_by: string | null;

  stage: ApplicantStage;

  tenant_ended_at: string | null;
  tenant_ended_reason: string | null;

  created_at: string;
  updated_at: string;

  screening_results: ScreeningResult[];
  references: ApplicantReference[];
  video_call_notes: VideoCallNote[];
  events: ApplicantEvent[];
}

File: apps/mybookkeeper/backend/app/schemas/applicants/applicant_detail_response.py
"""Pydantic schema for full Applicant detail responses.

Includes all 1:N children (screening_results, references, video_call_notes,
applicant_events) — used by GET /applicants/{id}. The detail page renders
the sensitive section behind a UI unlock toggle so PII isn't visible by
default per RENTALS_PLAN.md §9.1.

PII fields are returned plaintext via the ``EncryptedString`` TypeDecorator.
Auth-protected at the route layer.
"""
from __future__ import annotations

import datetime as _dt
import uuid

from pydantic import BaseModel, ConfigDict

from app.schemas.applicants.applicant_event_response import ApplicantEventResponse
from app.schemas.applicants.reference_response import ReferenceResponse
from app.schemas.applicants.screening_result_response import ScreeningResultResponse
from app.schemas.applicants.video_call_note_response import VideoCallNoteResponse


class ApplicantDetailResponse(BaseModel):
    id: uuid.UUID
    organization_id: uuid.UUID
    user_id: uuid.UUID
    inquiry_id: uuid.UUID | None = None

    legal_name: str | None = None
    dob: str | None = None
    employer_or_hospital: str | None = None
    vehicle_make_model: str | None = None
    id_document_storage_key: str | None = None

    contract_start: _dt.date | None = None
    contract_end: _dt.date | None = None
    smoker: bool | None = None
    pets: str | None = None
    referred_by: str | None = None

    stage: str

    tenant_ended_at: _dt.datetime | None = None
    tenant_ended_reason: str | None = None

    created_at: _dt.datetime
    updated_at: _dt.datetime

    screening_results: list[ScreeningResultResponse] = []
    references: list[ReferenceResponse] = []
    video_call_notes: list[VideoCallNoteResponse] = []
    events: list[ApplicantEventResponse] = []

    model_config = ConfigDict(from_attributes=True)
